net/postfix/src/opnsense/mvc/app/views/OPNsense/Postfix/general.volt

```diff
@@ -0,0 +1,106 @@
+    <li class="active"><a data-toggle="tab" href="#general">{{ lang._('General') }}</a></li>
+    <li><a data-toggle="tab" href="#antispam">{{ lang._('Antispam') }}</a></li>
+</ul>
+<div class="tab-content content-box tab-content">
+    <div id="general" class="tab-pane fade in active">
+        <div class="content-box" style="padding-bottom: 1.5em;">
+            {{ partial("layout_partials/base_form",['fields':generalForm,'id':'frm_general_settings'])}}
+            <hr />
+            <div class="col-md-12">
+                <button class="btn btn-primary"  id="saveAct" type="button"><b>{{ lang._('Save') }}</b><i id="saveAct_progress" class=""></i></button>
+            </div>
+        </div>
+    </div>
+    <div id="antispam" class="tab-pane fade in">
+        <div class="content-box" style="padding-bottom: 1.5em;">
+	    <div class="alert alert-warning" role="alert" id="missing_rspamd" style="display:none;min-height:65px;">
+                <div style="margin-top: 8px;">{{ lang._('No Rspamd plugin found, please install via System > Firmware > Plugins.')}}</div>
+            </div>
+            {{ partial("layout_partials/base_form",['fields':antispamForm,'id':'frm_antispam_settings'])}}
+            <hr />
+            <div class="col-md-12">
+                <button class="btn btn-primary"  id="saveAct2" type="button"><b>{{ lang._('Save') }}</b><i id="saveAct2_progress" class=""></i></button>
+            </div>
+        </div>
+    </div>
+</div>
+
+<script type="text/javascript">
+    $( document ).ready(function () {
+        var data_get_map = {'frm_general_settings':"/api/postfix/general/get"};
+        mapDataToFormUI(data_get_map).done(function (data) {
+            formatTokenizersUI();
+            $('.selectpicker').selectpicker('refresh');
+        });
+        var data_get_map2 = {'frm_antivirus_settings':"/api/postfix/antispam/get"};
+        mapDataToFormUI(data_get_map2).done(function(data){
+            formatTokenizersUI();
+            $('.selectpicker').selectpicker('refresh');
+        });
+        ajaxCall(url="/api/postfix/service/status", sendData={}, callback=function (data, status) {
+            updateServiceStatusUI(data['status']);
+        });
+
+	// check if Rspamd plugin is installed
+        ajaxCall(url="/api/postfix/service/checkrspamd", sendData={}, callback=function(data,status) {
+	    if (data == "0") {
+                $('#missing_rspamd').show();
+            }
+        });
+
+        // link save button to API set action
+        $("#saveAct").click(function () {
+            saveFormToEndpoint(url="/api/postfix/general/set", formid='frm_general_settings',callback_ok=function () {
+            $("#saveAct_progress").addClass("fa fa-spinner fa-pulse");
+                ajaxCall(url="/api/postfix/service/maketransport", sendData={});
+                ajaxCall(url="/api/postfix/service/reconfigure", sendData={}, callback=function (data,status) {
```

Comment:
I would not do that - I would recommend you to create the transport table in the reconfigure and start actions (if exist)

Reply:
ok, I moved it in the controller to the reconfigure action. 
Start is not required it think since with initial start it's not required. Will test with a fresh instance

Reply:
the transport map must exist when the server is starting. I would run postmap from the setup script as it is called anyway.

Reply:
Ah, good idea! 👍